src/tools.rs

```diff
@@ -243,6 +239,13 @@ pub fn guess_language<P: AsRef<Path>>(buf: &[u8], path: P) -> (Option<LANG>, Str
     }
 }
 
+pub(crate) fn remove_blank_lines(data: &mut Vec<u8>) {
+    while data.last().map_or_else(|| false, |c| *c == b'\n') {
```

Comment:
You could just count the trailing \n and then `resize` the vector correctly or push a new \n if no \n.